FUSETOOLS-2564 - WSDL 2 Camel Rest DSL Wizard

bfitzpat opened the pull request:
This is the first cut of the wsdl2rest code. The tests run after the jars are populated (mvn generate-sources) and the wizard attempts to run, but there's an issue with the wizard perform finish where it's not quite getting as far as the tests are. I suspect it has something to do with a classloader issue, but am still investigating. 

VERY PRELIMINARY WORK

bfitzpat commented:
Can now run the wsdl2rest utility code through the wizard. Working on updating the wizard to improve that experience next.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/src/org/jboss/fuse/wsdl2rest/ClassGenerator.java:
```diff
@@ -0,0 +1,9 @@
+package org.jboss.fuse.wsdl2rest;
```
missing header

bfitzpat commented:
Thanks for the feedback @apupier . I will add a readme file to help folks get some context on how to test this and will work through all of your suggestions. That said, this is merely a prototype branch and not meant to be the end-all-and-be-all, so until we figure out what we're doing with the core wsdl2rest code I am going to avoid splitting it out just yet.

lhein commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/internal/i18n/messages.properties:
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2017 Red Hat, Inc.
```
shouldn't the path be .../l10n/messages.properties?

bfitzpat replied:
Heh. Oops!

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/src/org/jboss/fuse/wsdl2rest/impl/ResourceMapperImpl.java:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    private boolean resourceExists(String resource) {
+        for (String str : this.resources) {
```
I think that resources.contains(resource); can be used

bfitzpat replied:
We will be pulling this code in jarred form from maven, so I won't change it at this time, but a good spot.

apupier replied:
I created a PR upstream https://github.com/jboss-fuse/wsdl2rest/pull/34

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/src/org/jboss/fuse/wsdl2rest/impl/ResourceMapperImpl.java:
```diff
@@ -0,0 +1,128 @@
+package org.jboss.fuse.wsdl2rest.impl;
```
missing header

bfitzpat replied:
We will be pulling this code in jarred form from maven, so I won't change it at this time, but a good spot.

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/pom.xml:
```diff
@@ -0,0 +1,71 @@
+		<dependency>
+			<groupId>commons-validator</groupId>
+			<artifactId>commons-validator</artifactId>
+			<version>1.4.0</version>
```
1.6.0 doesn't seem to be available in our maven repos currently, but this would be good to investigate

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/pom.xml:
```diff
@@ -0,0 +1,71 @@
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>ban-uberjars</id>
```
This was copied from the switchyard plug-ins (org.switchyard.tools.cxf was the model I used for the wsdl2rest code wrapper). Not sure we need it. Will investigate removing it when I start migrating to pulling wsdl2rest as a maven artifact instead of building a one-off copy of the code directly.

bfitzpat commented:
testPR

bfitzpat commented:
Neither of the failing tests has anything to do with my code. Any ideas @lhein or @apupier ? Just errant errors popping up randomly?

bfitzpat commented:
testPR

bfitzpat commented:
testPR

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,179 @@
+	        } finally {
+	            Thread.currentThread().setContextClassLoader(oldLoader);
+	        }
+		}
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java:
```diff
@@ -0,0 +1,386 @@
+            	Object rawLocation = 
+            			xpath.evaluate("/definitions/service/port/address/@location", document, XPathConstants.STRING);
+            	
+            	// if we found it, stash it!
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/.classpath:
```diff
@@ -0,0 +1,40 @@
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry exported="true" kind="lib" path="lib/args4j-2.0.31.jar"/>
```
this is still an impressive list of jars embedded in the plugin.
Do you have an idea of the size that it takes?
Do you have an idea how many of these dependencies are available on orbit or Jboss Tools Target Platform? (to evaluate the work to remove them)

bfitzpat replied:
We are limited here by the wsdl2rest tool itself. I simply brought in the jars it needed to operate successfully. So no, I have not looked at the size. And no, I have no idea what is available in orbit or the target platform. I suspect the versions alone will cause us issues even if some of these jars are available in orbit. But we had similar issues when we wrapped the CXF code for SwitchYard several years ago and suspect that the same limitations apply.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/.classpath:
```diff
@@ -0,0 +1,40 @@
+	<classpathentry exported="true" kind="lib" path="lib/args4j-2.0.31.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/javaparser-core-2.5.1.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/commons-collections-3.2.2.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/camel-spring-2.18.0.jar"/>
```
Does it means that we are bound to a specific Camel version?

bfitzpat replied:
The wsdl2rest tool is bound to a specific version, so I would suspect so.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/Stash/SpringCamelContextFactory.java:
```diff
@@ -0,0 +1,123 @@
+
+        @Override
+        public NamespaceHandler resolve(String namespaceUri) {
+            if ("http://camel.apache.org/schema/spring".equals(namespaceUri)) {
```
the constant org.fusesource.ide.foundation.core.xml.namespace.SpringNamespaceHandler.NAMESPACEURI_CAMEL_SPRING can be used

bfitzpat replied:
Unneeded file. Removed from project.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,218 @@
+
+  <artifactId>org.fusesource.ide.wsdl2rest</artifactId>
+  <packaging>eclipse-plugin</packaging>
+  <name>JBoss Fuse Tooling :: Camel Editor :: Plugins :: Wsdl2Rest Implemenatation </name>
```
typo Implemenatation -> Implementation

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,218 @@
+        <version.junit>4.11</version.junit>
+        <version.wsdl4j>1.6.3</version.wsdl4j>
+        <version.commons.lang>2.4</version.commons.lang>
+        <version.wsdl2rest>0.7-SNAPSHOT</version.wsdl2rest>
```
this version will be changed when a (productized?) release will be available, right?

bfitzpat replied:
Hope so, yes.

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.tests/src/org/jboss/fuse/wsdl2rest/test/doclit/CamelCxfWsDocLitTest.java:
```diff
@@ -0,0 +1,65 @@
+
+public class CamelCxfWsDocLitTest {
+
+//    @Test
```
completely commented out test class, Should we remove it or uncomment it?

bfitzpat replied:
Removed whole test plugin

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.tests/src/org/jboss/fuse/wsdl2rest/test/rpclit/CamelCxfWsRpcLitTest.java:
```diff
@@ -0,0 +1,65 @@
+
+public class CamelCxfWsRpcLitTest {
+
+//    @Test
```
completely commented out test class, shoudl we remove it or uncomment it?

bfitzpat replied:
Removed whole test plugin

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.tests/src/org/jboss/fuse/wsdl2rest/test/rpclit/GenerateRpcLitTest.java:
```diff
@@ -0,0 +1,74 @@
+        Assert.assertEquals(5, methods.size());
+    }
+
+//    @Test
```
is this commented out test relevant?
if yes, can you remove it please
If no, can you enable it please

bfitzpat replied:
Removed whole test plugin

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/readme.md:
```diff
@@ -0,0 +1,39 @@
+On the "Specify Advanced Options for wsdl2rest Processing", click the "..." beside the Bean Class field and select the EchoServiceImpl class. 
+Click Finish.
+
+Select the project and refresh it to see the generated code and new rest-camel-context.xml file. 
```
I say "...to see the generated code and new rest-camel-context.xml file." The second "the" you included is implied because it's a simple list of two items. You could certainly include it, but it's not necessary.

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/readme.md:
```diff
@@ -0,0 +1,39 @@
+
+The generated echo.Echo class will have many errors due to some missing dependencies.
+
+Add the following dependencies to your project pom.xml:
```
Created https://issues.jboss.org/browse/FUSETOOLS-2869

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardFirstPage.java:
```diff
@@ -0,0 +1,134 @@
+		IPath javaPath = new Path(path3);
+		String projectJavaPath = selectedProject.getFullPath().append(javaPath).toPortableString();
+		String projectConfigPath = null;
+		if (isProjectBlueprint()) {
```
We don't do Java DSL currently. Just Spring & Blueprint. So good corner case to ponder.

cunningt commented:
testPR

bfitzpat commented:
Next changes include attempting to add required libs so that generated code builds successfully. But this update included some updates to work with upstream changes in wsdl2rest that eliminates the optional need for a bean class and adds the ability to customize the rest target address.

apupier commented:
I set rebase required because the diff contains all the Jboss Fuse --> red Hat Fuse modifications which is making hard to see what has been effectively modified

bfitzpat commented:
That may have been the ugliest merge I have ever seen. Not sure why. But waiting on a build

bfitzpat commented:
Thanks for the feedback @apupier  - will take a look and get back to you

apupier commented:
the 2 wsdl2rest plugins are not included in the build, they need to be part of the editor/plugins/pom.xml
it reminds me that they need to be declared in the features.xml too

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/readme.md:
```diff
@@ -0,0 +1,22 @@
+1. Create a new Fuse Integration project. Leave it as a Blank project that uses the Spring DSL.
+2. Create a new src/main/resources/wsdl directory.
+3. Copy the HelloService.wsdl file into the src/main/resources/wsdl directory.
+4. Select the project, click New->Other, and select the JBoss Fuse->WSDL to Camel Rest DSL Wizard. 
```
Couple of things on the commons-validator dependency. Tried 1.6 - does not appear to be available in the maven repos we pull from at this point. 

As for getting an orbit bundle - I spoke to Nick Boldt about that - he says we need to reach out to  Roland Grunberg, aka, The Orbit King about possibly adding apache commons-validator (https://commons.apache.org/proper/commons-validator/) to Orbit. From Nick - "get in RH IRC #eclipse. look for rgrunberg. ping with impunity" :)

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,269 @@
+		try {
+			generate();
+		} catch (Exception e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
```
Maybe?

apupier replied:
yes, maybe. To be discussed. What do you think? Better to do it in this initial iteration or in a following ones?

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,218 @@
+            <artifactId>commons-lang</artifactId>
+        </dependency>
+
+        <!-- Test -->
```
Will remove. Was a remnant from an earlier iteration.

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,218 @@
+              <goal>copy-dependencies</goal>
+            </goals>
+            <configuration>
+              <excludeScope>system</excludeScope>
```
Copied from switchyard - don't remember why. Maybe we don't need it?

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,150 @@
+	 * @param wsdlURL
+	 */
+	public void setWsdlURL(String wsdlURL) {
+		firePropertyChange("wsdlURL", this.wsdlURL, this.wsdlURL = wsdlURL); //$NON-NLS-1$
```
Have not tried without, but it seems to work ok so why change?

bfitzpat commented:
Where do I need to include them in the build or which feature.xml should I modify to include those projects?

apupier commented:
> Where do I need to include them in the build or which feature.xml should I modify to include those projects?

to include in the build, need to add in the parent pom.xml like for classical Maven projects.
Put in feature.xml associated to the folder you created the plugins, so the editor one in this case.

See https://github.com/jbosstools/jbosstools-fuse/pull/1138/commits/b1ce1117c86cd8c49752f37ae6ebae87187742db

Next part when including new plugins is that we need to duplicate this information in some jboss tools places after the plugins has been effectively added (so PR merged and master published with changed) I created this subtask for that: https://issues.jboss.org/browse/FUSETOOLS-2922

apupier commented:
I had a look to the commit history and almost none of them contains the issue number, please add it

apupier commented:
unless I missed something, the build cannot work for now as API for a version not productized yet is used and there is no public -SNAPSHOT artefacts.
the built versions have changing version number and they are placed on staging repositories:
![image](https://user-images.githubusercontent.com/1105127/38620648-4792e89e-3d9f-11e8-8508-f9e4cdc8d5ea.png)

bfitzpat commented:
I have not seen a new productized version of wsdl2rest since the 0.7.0-000017.fuse-000001 version you found on Nexus. That said, to build locally you can clone the wsdl2rest repo locally - https://github.com/jboss-fuse/wsdl2rest - build that in your local maven repo, and then everything should build ok on your box (it does for me). 

I'm hopeful that when they build the artifacts for the release we'll get a new build, but I'll check with Tom Cunningham to see what's going on there.

bfitzpat commented:
Unfortunately until we get that non-snapshot version, the plug-ins won't build due to the constraint: 

```
[ERROR] Found property version.wsdl2rest = 0.7-SNAPSHOT
[WARNING] Rule 0: org.jboss.tools.releng.NoSnapshotsAllowed failed with message:

When buildAlias (Final) matches /Final|GA/, cannot include SNAPSHOT dependencies.

```

apupier commented:
provided proposal for improvement: https://github.com/bfitzpat/fuseide/pull/17

bfitzpat commented:
Hey @apupier I'll look at the copy vs. copy dependencies fix next, but I moved the commons-validator dependency up to the wsdl2rest wrapper for now (figured it was better to have all the jars in one place). This last update also includes a version change so we're using a productized version and I added an integration test.

bfitzpat commented:
And I'm seeing these errors with the build, so I'm not sure what's up. Any ideas?
```
[INFO] --- tycho-packaging-plugin:1.1.0:package-plugin (default-package-plugin) @ org.fusesource.ide.wsdl2rest ---
[WARNING] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest\build.properties: bin.includes value(s) [lib/cxf-codegen-plugin-3.2.1.jar, lib/javaparser-core-2.5.1.jar, lib/velocity-1.7.jar, lib/args4j-2.33.jar, lib/javax.ws.rs-api-2.1.jar, lib/cxf-tools-common-3.2.1.jar, lib/cxf-tools-wsdlto-core-3.2.1.jar, lib/cxf-tools-wsdlto-frontend-jaxws-3.2.1.jar, lib/stax2-api-3.1.4.jar, lib/wsdl4j-1.6.3.jar, lib/commons-collections-3.2.2.jar, lib/cxf-tools-wsdlto-databinding-jaxb-3.2.1.jar, lib/jaxb-xjc-2.2.11.jar, lib/xml-resolver-1.2.jar, lib/cxf-tools-validator-3.2.1.jar, lib/ant-1.10.1.jar, lib/ant-launcher-1.10.1.jar, lib/ant-nodeps-1.8.1.jar, lib/cxf-tools-wsdlto-frontend-javascript-3.2.1.jar, templates/, lib/commons-lang-2.4.jar, lib/wsdl2rest-util.jar, lib/wsdl2rest-impl.jar, lib/woodstox-core-5.0.3.jar, lib/camel-spring-2.21.0.fuse-000069.jar, lib/cxf-core-3.1.11.fuse-000226.jar, lib/cxf-codegen-plugin-3.1.11.fuse-000226.jar, lib/cxf-rt-databinding-jaxb-3.1.11.fuse-000226.jar, lib/cxf-rt-wsdl-3.1.11.fuse-000226.jar, lib/cxf-tools-common-3.1.11.fuse-000226.jar, lib/cxf-tools-wsdlto-core-3.1.11.fuse-000226.jar, lib/cxf-tools-wsdlto-frontend-jaxws-3.1.11.fuse-000226.jar, lib/slf4j-api-1.7.25.jar, lib/woodstox-core-asl-4.4.1.jar, lib/jaxb-core-2.3.0.jar, lib/cxf-tools-wsdlto-databinding-jaxb-3.1.11.fuse-000226.jar, lib/xmlschema-core-2.2.1.jar, lib/cxf-tools-validator-3.1.11.fuse-000226.jar, lib/cxf-tools-wsdlto-frontend-javascript-3.1.11.fuse-000226.jar, lib/cxf-rt-bindings-soap-3.1.11.fuse-000226.jar, lib/commons-validator-1.6.jar] do not match any files.
[INFO] Building jar: E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest\target\org.fusesource.ide.wsdl2rest-10.3.0-SNAPSHOT.jar
```

bfitzpat commented:
`[ERROR] Failed to execute goal on project org.fusesource.ide.wsdl2rest.ui: Could not resolve dependencies for project org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.ui:eclipse-plugin:10.3.0-SNAPSHOT: The following artifacts could not be resolved: org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/javaparser-core-2.5.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/commons-collections-3.2.2.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/stax2-api-3.1.4.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/commons-lang-2.4.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/velocity-1.7.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/wsdl4j-1.6.3.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/jaxb-xjc-2.2.11.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/xml-resolver-1.2.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/ant-nodeps-1.8.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/args4j-2.33.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/camel-spring-2.21.0.fuse-000069.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-core-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-codegen-plugin-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-rt-databinding-jaxb-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-rt-wsdl-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-common-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-wsdlto-core-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-wsdlto-frontend-jaxws-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/javax.ws.rs-api-2.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/slf4j-api-1.7.25.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/woodstox-core-asl-4.4.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/jaxb-core-2.3.0.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-wsdlto-databinding-jaxb-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/xmlschema-core-2.2.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-validator-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/ant-1.10.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/ant-launcher-1.10.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-tools-wsdlto-frontend-javascript-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/cxf-rt-bindings-soap-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/wsdl2rest-util.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/commons-validator-1.6.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/javaparser-core-2.5.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/commons-collections-3.2.2.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/stax2-api-3.1.4.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/commons-lang-2.4.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/velocity-1.7.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/wsdl4j-1.6.3.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/jaxb-xjc-2.2.11.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/xml-resolver-1.2.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/ant-nodeps-1.8.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/args4j-2.33.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/camel-spring-2.21.0.fuse-000069.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-core-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-codegen-plugin-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-rt-databinding-jaxb-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-rt-wsdl-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-common-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-wsdlto-core-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-wsdlto-frontend-jaxws-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/javax.ws.rs-api-2.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/slf4j-api-1.7.25.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/woodstox-core-asl-4.4.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/jaxb-core-2.3.0.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-wsdlto-databinding-jaxb-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/xmlschema-core-2.2.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-validator-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/ant-1.10.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/ant-launcher-1.10.1.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-tools-wsdlto-frontend-javascript-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/cxf-rt-bindings-soap-3.1.11.fuse-000226.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/wsdl2rest-util.jar:10.3.0.qualifier, org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:jar:lib/commons-validator-1.6.jar:10.3.0.qualifier: Could not find artifact org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest.source:jar:lib/javaparser-core-2.5.1.jar:10.3.0.qualifier at specified path E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest\lib\javaparser-core-2.5.1.jar -> [Help 1]
`

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,36 @@
+Bundle-ManifestVersion: 2
+Bundle-Name: Fuse wsdl2rest UI Plugin
+Bundle-SymbolicName: org.fusesource.ide.wsdl2rest.ui;singleton:=true
+Bundle-Version: 10.3.0.qualifier
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/pom.xml:
```diff
@@ -0,0 +1,15 @@
+	<parent>
+		<groupId>org.fusesource.ide.editor</groupId>
+		<artifactId>plugins</artifactId>
+		<version>10.3.0-SNAPSHOT</version>
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,49 @@
+Bundle-ManifestVersion: 2
+Bundle-Name: Fuse wsdl2rest Wrapper Plugin
+Bundle-SymbolicName: org.fusesource.ide.wsdl2rest
+Bundle-Version: 10.3.0.qualifier
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,222 @@
+  <parent>
+    <groupId>org.fusesource.ide.editor</groupId>
+    <artifactId>plugins</artifactId>
+    <version>10.3.0-SNAPSHOT</version>
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.ui.tests.integration/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,24 @@
+Bundle-ManifestVersion: 2
+Bundle-Name: %Bundle-Name
+Bundle-SymbolicName: org.fusesource.ide.wsdl2rest.ui.tests.integration
+Bundle-Version: 10.3.0.qualifier
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.ui.tests.integration/pom.xml:
```diff
@@ -0,0 +1,16 @@
+  <parent>
+    <groupId>org.fusesource.ide.editor</groupId>
+    <artifactId>tests</artifactId>
+    <version>10.3.0-SNAPSHOT</version>
```
as version has been upgraded will need to modify to 11.0.0 (and rebase on master)

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.ui.tests.integration/src/test/java/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizardTest.java:
```diff
@@ -0,0 +1,113 @@
+	    IResourceVisitor webInfCollector = new IResourceVisitor() {
+	      @Override
+	      public boolean visit(IResource resource) throws CoreException {
+				System.out.println("Resource visitor is visiting " + resource.toString());
```
remind to remove this sysout

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.ui.tests.integration/src/test/java/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizardTest.java:
```diff
@@ -0,0 +1,113 @@
+		Iterator<IFile> fileIter = files.iterator();
+		while (fileIter.hasNext()) {
+			IFile tempFile = fileIter.next();
+			System.out.println(tempFile.toString());
```
don't forget to remove the sysout

bfitzpat commented:
Still have build issues, but at least versions jive with master now that the branch is up to date.

apupier commented:
based on erro in build:
[ERROR] Failed to execute goal on project org.fusesource.ide.wsdl2rest: Could not resolve dependencies for project org.fusesource.ide.editor:org.fusesource.ide.wsdl2rest:eclipse-plugin:11.0.0-SNAPSHOT: Failed to collect dependencies at org.apache.camel:camel-spring:jar:2.21.0.fuse-000069: Failed to read artifact descriptor for org.apache.camel:camel-spring:jar:2.21.0.fuse-000069: Could not transfer artifact org.apache.camel:camel-spring:pom:2.21.0.fuse-000069 from/to central (https://repository.jboss.org/nexus/content/repositories/central/): Connection reset -> [Help 1]

I think that the "promoted repository" is at a specific url that needs to be added to the pom.xml

apupier commented:
provided PR to fix test plugin: https://github.com/bfitzpat/fuseide/pull/19 (with the test failing)

apupier commented:
another PR to let the build find the embedded dependencies https://github.com/bfitzpat/fuseide/pull/20

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,294 @@
+						tool.setCamelContext(contextpath); 
+					} else {
+						IPath projectPath = ResourcesPlugin.getWorkspace().getRoot().findMember(project.getName()).getLocation();
+						Path contextpath = new File(projectPath.makeAbsolute() + File.separator + "rest-blueprint-context.xml").toPath(); //$NON-NLS-1$
```
this variable is not used. Is it normal? If normal, the whole line can be removed

bfitzpat replied:
Fixed

bfitzpat commented:
Merged the two additional PRs (thank you) and fixed the errant line where I wasn't using the variable that was defined.

apupier commented:
provided PR to fix wizard which is no more opening: https://github.com/bfitzpat/fuseide/pull/21

bfitzpat commented:
Wait - got a different result. Looks like the wsdl2rest libraries aren't being exposed properly via the wrapper...
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.1.0:compile (default-compile) on project org.fusesource.ide.wsdl2rest.ui: Compilation failure: Compilation failure:
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[13]
[ERROR] import org.apache.commons.validator.routines.UrlValidator;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The import org.apache.commons.validator cannot be resolved
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a variable
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[35]
[ERROR] if (!urlValidator.isValid((String)value)) {
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[48]
[ERROR] import org.jboss.fuse.wsdl2rest.impl.Wsdl2Rest;
[ERROR] ^^^^^^^^^^^^^^
[ERROR] The import org.jboss.fuse cannot be resolved
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[235]
[ERROR] Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
[ERROR] ^^^^^^^^^
[ERROR] Wsdl2Rest cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[235]
[ERROR] Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
[ERROR] ^^^^^^^^^
[ERROR] Wsdl2Rest cannot be resolved to a type
[ERROR] 8 problems (8 errors)
[ERROR] -> [Help 1]
```

bfitzpat commented:
Closer though!

apupier commented:
updated PR which should fix the missing WSDl2Rest class

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,294 @@
+			generate();
+		} catch (Exception e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
+			MessageDialog.openError(getShell(), "Error", e.getMessage());
```
given that the iexception thrown is an encapsulated InvocationTargetException thrown by the Wsdl2Rest.generate method, the message will be always empty.

![image](https://user-images.githubusercontent.com/1105127/38995824-17895ef2-43ea-11e8-8012-6e1b212489de.png)

either improve the message in the encapsulated exception or just throw the original exception

apupier replied:
might be also a good idea to use the org.eclipse.jface.dialogs.ErrorDialog to have the stacktrace visible in UI if user wants to expand details.

apupier replied:
I provided a PR for this branch https://github.com/bfitzpat/fuseide/pull/23

bfitzpat commented:
Ok, I moved the repository references into the base pom for the fuseide project and get a little further, but it doesn't seem that the wsdl2rest jars are being exposed on the classpath properly. 

I added the same repos you did in the wsdl2rest wrapper project and also added them as pluginRepositories just to see if that helped. 

It leads to these failures:
```
 [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.1.0:compile (default-compile) on project org.fusesource.ide.wsdl2rest.ui: Compilation failure: Compilation failure:
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[13]
[ERROR] import org.apache.commons.validator.routines.UrlValidator;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The import org.apache.commons.validator cannot be resolved
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[27]
[ERROR] UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a variable
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\pages\TargetURLValidator.java:[35]
[ERROR] if (!urlValidator.isValid((String)value)) {
[ERROR] ^^^^^^^^^^^^
[ERROR] UrlValidator cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[47]
[ERROR] import org.jboss.fuse.wsdl2rest.impl.Wsdl2Rest;
[ERROR] ^^^^^^^^^^^^^^
[ERROR] The import org.jboss.fuse cannot be resolved
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[197]
[ERROR] Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
[ERROR] ^^^^^^^^^
[ERROR] Wsdl2Rest cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[197]
[ERROR] Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
[ERROR] ^^^^^^^^^
[ERROR] Wsdl2Rest cannot be resolved to a type
[ERROR] E:\Users\bfitzpat\github\fuseide\editor\plugins\org.fusesource.ide.wsdl2rest.ui\src\org\fusesource\ide\wsdl2rest\ui\wizard\Wsdl2RestWizard.java:[223]
[ERROR] protected void initContextForTool(boolean isBlueprint, IPath camelPath, File camelFile, Wsdl2Rest tool) {
[ERROR] ^^^^^^^^^
[ERROR] Wsdl2Rest cannot be resolved to a type
[ERROR] 9 problems (9 errors)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.

```

apupier commented:
> but it doesn't seem that the wsdl2rest jars are being exposed on the classpath properly.

They are exposed properly on the classpath but they are not downloaded in lib folder when launching the build from top-level pom

apupier commented:
please see https://github.com/bfitzpat/fuseide/pull/22 which is fixing/avoiding all Eclipse warnings

apupier commented:
if there is no project selected initially, there is an erro when clicking finish (even if the project has been correctly selected in UI)
there is this stacktrace:
!ENTRY org.fusesource.ide.wsdl2rest.ui 4 0 2018-04-20 11:17:40.746
!MESSAGE <no message>
!STACK 0
java.lang.NullPointerException
	at org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils.getMavenProjectFacade(CamelMavenUtils.java:103)
	at org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils.getDependencyList(CamelMavenUtils.java:194)
	at org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils.getDependencyList(CamelMavenUtils.java:190)
	at org.fusesource.ide.wsdl2rest.ui.wizard.Wsdl2RestWizard.isProjectBlueprint(Wsdl2RestWizard.java:134)
	at org.fusesource.ide.wsdl2rest.ui.wizard.Wsdl2RestWizard.generate(Wsdl2RestWizard.java:184)
	at org.fusesource.ide.wsdl2rest.ui.wizard.Wsdl2RestWizard.performFinish(Wsdl2RestWizard.java:84)
	at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:778)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:417)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:81)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at org.eclipse.ui.internal.handlers.WizardHandler$New.executeHandler(WizardHandler.java:269)
	at org.eclipse.ui.internal.handlers.WizardHandler.execute(WizardHandler.java:290)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:291)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:305)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:239)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
	at org.eclipse.ui.internal.handlers.LegacyHandlerService.executeCommand(LegacyHandlerService.java:390)
	at org.eclipse.ui.internal.actions.CommandAction.runWithEvent(CommandAction.java:151)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:565)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:397)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1472)

apupier commented on core/tests/org.fusesource.ide.camel.model.tests/.project:
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
see https://github.com/bfitzpat/fuseide/pull/25

apupier commented on editor/tests/org.fusesource.ide.wsdl2rest.ui.tests.integration/src/main/java/org/fusesource/ide/wsdl2rest/ui/tests/integration/Wsdl2RestWizardIT.java:
```diff
@@ -0,0 +1,112 @@
+	public void testWsdl2RestWizardFinish() throws Exception {
+    	Wsdl2RestWizard wizard = new Wsdl2RestWizard();
+        File wsdlFile = new File(WSDL_LOCATION);
+        Path outpath = new File(OUTPUT_PATH).toPath();
```
indentation issue, please merge first https://github.com/bfitzpat/fuseide/pull/24 to avoid potential conflict

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,237 @@
+  <repositories>
+    	<repository>
+  		<id>Saved staging repo for cxf</id>
+  		<url>https://repository.jboss.org/nexus/content/repositories/fusesource_releases_external-9109/</url>
```
comment to not forget to be sure that all the repository are saved before we do a merge in master

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/readme.md:
```diff
@@ -0,0 +1,22 @@
+Once the wizard finishes, you should see a new Camel Context file - rest-camel-context.xml - in the src/main/resources directory as well as the generated code, which will appear in the /src/main/java folder.
+
+## Current Limitations
+* Only works with Spring DSL projects currently.
```
I think the readme is no more up to date

apupier commented:
> 15:10:23 [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-source-feature-plugin:1.1.0:source-feature (source-feature) on project org.fusesource.ide.camel.editor.feature: Could not generate source feature for project MavenProject: org.fusesource.ide.editor:org.fusesource.ide.camel.editor.feature:11.0.0-SNAPSHOT @ /home/jenkins/workspace/jbosstools-fuse_PullRequest/editor/features/org.fusesource.ide.camel.editor.feature/pom.xml
> 15:10:23 [ERROR]     Missing sources for plugins [org.fusesource.ide.wsdl2rest_11.0.0.v20180420-1312]

pfouh update site requires to have some "src" configured even if there is no class

bfitzpat commented:
Interesting. Is that the definition of the src folder in the plugin itself or something to do with the feature definition?

bfitzpat commented:
Looks like maybe we can exclude the plug-in from source generation based on this https://bugs.eclipse.org/bugs/show_bug.cgi?id=389762#c14

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/internal/Wsdl2RestUIActivator.java:
```diff
@@ -0,0 +1,75 @@
+	@Override
+	public void start(BundleContext context) throws Exception {
+		super.start(context);
+		Wsdl2RestUIActivator.plugin = this;
```
False positive I think

bfitzpat commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java:
```diff
@@ -0,0 +1,348 @@
+	 * Opens a simple dialog to allow selection of a WSDL file.
+	 */
+	protected void selectWSDL() {
+		FilteredResourcesSelectionDialog dialog = new FilteredResourcesSelectionDialog(getShell(), false,
```
False positive I think

apupier commented:
- to avoid having a "dirty" editor when opening the generated route, it would be better to force generation of the id in the "perform finish" of the wizard. This is something that might be done in a following PR. if you think so, please open a JIRA subtask.

apupier commented:
the destination Camel foldr has not been respected for me, with
![image](https://user-images.githubusercontent.com/1105127/39115528-32b2af80-46e2-11e8-8c3b-049cc9ca1126.png)

the file rest-camel-context.xml has been created at the root of the project

apupier commented:
in case the user is trying to generate in a project which already contains a camel context file with same name or same Java classes, everything is overridden without asking user. This can be considered as data loss from user point of view (imagine he/she mistakenly choose the wrong project)

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java:
```diff
@@ -0,0 +1,348 @@
+					locationURL = (String) rawLocation;
+				}
+			} catch (IOException | ParserConfigurationException | SAXException | XPathExpressionException e) {
+				// should be valid at this point
```
what do you mean by "should be valid at this point"? although it is in a catch exception block?

bfitzpat replied:
I mean we shouldn't hit this if there's a URL specified in the WSDL. If not, then we have to deal with that case.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java:
```diff
@@ -0,0 +1,346 @@
+					@Override
+					public boolean matchItem(Object item) {
+						IResource resource = (IResource) item;
+						return super.matchItem(item) && "wsdl".equals(resource.getFileExtension()); //$NON-NLS-1$
```
it is when using a FileDialog, please see https://github.com/bfitzpat/fuseide/pull/30 for a different proposal.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param wsdlURL
+	 */
+	public void setWsdlURL(String wsdlURL) {
+        String oldValue = this.wsdlURL;
```
indentation with space

bfitzpat commented:
Created https://issues.jboss.org/browse/FUSETOOLS-2945 to spin off ID generation during the performFinish.

bfitzpat commented:
"the file rest-camel-context.xml has been created at the root of the project" - did that folder exist ahead of time? Or not?

apupier commented:
> "the file rest-camel-context.xml has been created at the root of the project" - did that folder exist ahead of time? Or not?

yes, the folder exists ahead of time. I reproduce only with Spring

![image](https://user-images.githubusercontent.com/1105127/39128231-8adfda6e-4707-11e8-8838-93f691f0f63e.png)
![image](https://user-images.githubusercontent.com/1105127/39128567-6321362a-4708-11e8-811f-f4a3814cdd56.png)
![image](https://user-images.githubusercontent.com/1105127/39128584-6c28f42e-4708-11e8-9608-e25a2a321670.png)
![image](https://user-images.githubusercontent.com/1105127/39128626-87db720a-4708-11e8-85d1-e0f1b5cdd0b4.png)

bfitzpat commented:
After really testing that File Dialog, I'm not liking having to go outside Eclipse to find a WSDL path. We're tossing them to the root of the file system now vs. looking for a WSDL file in the project that they're looking to generate inside.

bfitzpat commented:
But I am able to reproduce the issue with not generating the config in the right place. Investigating now.

bfitzpat commented:
Figured out why it's generating the file in the root and not the folder specified. That folder doesn't exist. It's a Springboot project, which has src/main/resources/spring instead of src/main/resources/META-INF/spring. 

We either need to validate that folder a bit better (definitely do) or figure out a better way to figure out the starting path for a new camel configuration based on the project selected. 

@apupier

bfitzpat commented:
Last commit adds better validation to the path fields (adding a warning if the path doesn't exist yet and can be created or an error if there's no way to create it, which shouldn't be the case) and adds the camel path if it needs to. 

I still think there should be a way to introspect the camel project and get the actual location (i.e. different between springboot, spring, and blueprint projects) and we could just use that to avoid having to create the folder.

apupier commented:
test is failing on CI:
```
java.lang.AssertionError: 
Expecting:
 <false>
to be equal to:
 <true>
but was not.
	at org.fusesource.ide.wsdl2rest.ui.tests.integration.Wsdl2RestWizardIT.testWsdl2RestWizardFinish(Wsdl2RestWizardIT.java:61)
```

apupier commented:
> I still think there should be a way to introspect the camel project and get the actual location (i.e. different between springboot, spring, and blueprint projects) and we could just use that to avoid having to create the folder.

maybe we can start by having a different path for springboot/spring/blueprint which are matching the templates that we are providing in Tooling?

apupier commented:
> After really testing that File Dialog, I'm not liking having to go outside Eclipse to find a WSDL path. We're tossing them to the root of the file system now vs. looking for a WSDL file in the project that they're looking to generate inside.

it is going to the default path registered on the OS.
It is also possible to set an initial path, we can put the currently selected project for instance.

bfitzpat commented:
In the process of work on making sure we're not overwriting files, it became clear that a bit more clean-up needed to happen.

* removed duplicated "find resource" method and streamlined
* added new tests to ensure Blueprint path is working
* improved FuseProject class with better Blueprint support

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/META-INF/MANIFEST.MF:
```diff
@@ -13,7 +13,7 @@ Require-Bundle: org.eclipse.ui,
  org.jboss.tools.foundation.core,
  org.jboss.tools.foundation.ui,
- org.fusesource.ide.foundation.core,
+ org.fusesource.ide.foundation.core;visibility:=reexport,
```
better to always avoid to "reexport", if another plugin needs the dependencies, it should declare the dependencies in its MANIFEST too

apupier commented:
> Do we have a utility to discover if a project is Springboot vs. Spring vs. Blueprint?

Not exactly
There is org.fusesource.ide.camel.editor.utils.MavenUtils.isSyndesisExtensionProject(IProject) which might make sense to factorize code with.
There are org.fusesource.ide.foundation.core.util.CamelUtils.isBlueprintFile(String) and org.fusesource.ide.foundation.core.util.CamelUtils.isSpringFile(String) to detect File type
there are org.fusesource.ide.camel.model.service.core.util.CamelCatalogUtils.hasSpringBootDependency(List<Dependency>)
org.fusesource.ide.launcher.run.util.MavenLaunchUtils.isSpringBootProject(IFile)

surely some code to reuse/factorize :-)

apupier commented:
some tests are failing, I guess due to modification of the FuseProject rule:
![image](https://user-images.githubusercontent.com/1105127/39352636-9d232608-4a05-11e8-88a5-b2202c7975b1.png)

bfitzpat commented:
Thanks for the hints as to springboot detection. 

And damn. Sorry about the FuseProject change. I'll see what I can figure out and get working again.

bfitzpat commented:
I believe I have fixed the issue with FuseProject - it doesn't fail locally now in those Red deer tests

lhein commented:
also this PR is not rebased with latest master and still on version 10.x

bfitzpat commented:
This one is now obsolete. We are going with https://github.com/jbosstools/jbosstools-fuse/pull/1362

bfitzpat commented:
Can close once we merge the other PR
